contracts/L1/gateway/L1Migrator.sol

```diff
@@ -214,6 +242,62 @@ contract L1Migrator is L1ArbitrumMessenger, IMigrator, EIP712 {
         emit MigrateSenderInitiated(seqNo, params);
     }
 
+    // TODO: Add whenNotPaused modifier to prevent this function from being called until other contracts are ready
+    function migrateETH(
+        uint256 _maxGas,
+        uint256 _gasPriceBid,
+        uint256 _maxSubmissionCost
+    ) external payable {
+        require(!ethMigrated, "L1Migrator#migrateETH: ALREADY_MIGRATED");
+
+        ethMigrated = true;
+
+        uint256 amount = IBridgeMinter(bridgeMinterAddr)
+            .withdrawETHToL1Migrator();
+
+        // Any ETH refunded to the L2 alias of this contract can be used for
+        // other cross-chain txs sent by this contract.
+        // The retryable ticket created will not be cancellable since this contract
+        // currently does not support cross-chain txs to call ArbRetryableTx.cancel().
+        sendTxToL2(
+            l2MigratorAddr,
+            address(this), // L2 alias of this contract will receive refunds
```

Comment:
> can be used for other cross-chain txs sent by this contract

Wouldn't the other txs also contain attached eth anyway?

Reply:
The other cross-chain txs (i.e. in `migrateDelegator()`) set the credit back address & beneficiary to the `_l2Addr` specified by the caller of the migration function. As a result, `_l2Addr` will:

1. Receive refunds on L2 for the base submission as well as L2 execution as the credit back address
2. Receive refunds on L2 for the L2 call value if the ticket is timed out/cancelled as the beneficiary
3. Be authorized to cancel the ticket as the beneficiary

Assuming that the ticket is executed, then the refunds are for point 1. And for point 1, `_l2Addr` receives the refund which does not affect any other addresses that trigger a migration function in L1Migrator.

However, if we set the credit back address and beneficiary to the L1Migrator here then the L2 alias of the L1Migrator will receive the refund for point 1. When the next address that triggers a migration function, the L2 alias of the L1Migrator will have some amount of ETH on L2 sitting in its balance already from the previous refund. So, that balance can actually be used to cover some of the L2 execution cost since the L2 alias of the L1Migrator will be the sender of the tx on L2.

I think one of the reasons I structured it this way was so the "from" for this cross-chain tx would show up as the L1Migrator as opposed to the caller of the function since the L1Migrator is where the ETH is coming from as opposed to the caller.

There might've been other reasons that I am blanking on right now - will think on it, but happy to hear any arguments for using msg.sender.

Reply:
Yeah that sounds reasonable 
msg.sender receiving callvalue in case of point 2 is **absolutely not** desirable.

Reply:
Yeah that is really really important to highlight. We cannot set msg.sender as the beneficiary here because otherwise the msg.sender could call `migrateETH()` which sets the L2 call value as the amount pulled from the BridgeMinter and then msg.sender could cancel the ticket before it is executed on L2 to receive that amount.

Reply:
Added a comment on the above in 3716b04